src/frontend/components/contribute/contribute-start.tsx

```diff
@@ -26,13 +27,14 @@ export class ContributeStart extends React.Component<IContributeStartProps, any>
                         {listText}
                     </li>)}
                 </ul>
-                <ProveYouAreHuman onSuccess={ ContributeStart.onCaptchaSuccess } />
+                <ProveYouAreHuman onSuccess={ this.onCaptchaSuccess } />
             </div>
         );
     }
 
-    static onCaptchaSuccess(sessionToken: string) {
-        setState( { sessionToken });
+    private onCaptchaSuccess(result) {
+        if(this.props.state.sessionToken) return;
```

Comment:
This should be a single function that changes all state at once (streamlines state) It's called contributeStartCaptchaSuccess and it's in the "state" folder.